### PR TITLE
KIALI-1975 Runtime monitoring: list available runtimes for app and workload

### PR DESCRIPTION
**Pull request by jotak**
JIRA: https://issues.jboss.org/browse/KIALI-1975

- Add CustomDashboards in app & workload details
- Get dashboards by parsing annotation kiali.io/dashboards on pods
- Memoize title to avoid reloading resource to often

### CODE REVIEW

**Comment by jotak** [comment]
DNM label: this PR has still some TODOs inside... I'll need #726 to be merged first before I can finalize this one.

**Comment by jotak** [comment]
@xeviknal @lucasponce this is ready for review. It allows to match pod annotations with dashboards, and fill a list the that the UI can create tabs per dashboard.

**Comment by lucasponce** [comment]
QE JIRA for this https://issues.jboss.org/browse/KIALI-2054

File: business/dashboards.go [hunk 1]
```diff
@@ -148,3 +169,30 @@ func (in *DashboardsService) GetIstioDashboard(params prometheus.IstioMetricsQue
...
+	key := dashboardKey(namespace, template)
+	if title, ok := dashboardTitles[key]; ok {
+		return title
+	}
+	dashboard, err := in.loadDashboardResource(namespace, template)
+	if err != nil {
+		log.Errorf("Cannot get dashboard %s in namespace %s", template, namespace)
+		return ""
+	}
+	return dashboard.Spec.Title
```
Comment (lucasponce) [inline business/dashboards.go]: Instead to return the dashboard.Spec.title, why not to fill the dashboardTitles and return always the cache ?

Comment (jotak) [reply business/dashboards.go]: `dashboardTitles` is filled anyway via `in.loadDashboardResource`, so it isn't really different from what you say. I can rename it to make it clearer it's updating the cache.

File: business/workloads.go [hunk 2]
```diff
@@ -989,3 +991,23 @@ func controllerPriority(type1, type2 string) string {
...
+			}
+		}
+	}
+	mon, err := kubernetes.NewKialiMonitoringClient()
+	if err != nil {
+		// Do not fail the whole query, just log & return
+		log.Error("Cannot initialize Kiali Monitoring Client")
+		return
+	}
+	dash := NewDashboardsService(mon, in.prom)
```
Comment (lucasponce) [inline business/workloads.go]: Side comment.
Probably we want to merge all these XYZService split into one.
With the level of interdependencies we can simplify them, but of course, it has nothing to do with this PR, just to not forget.

File: models/pod.go [hunk 3]
```diff
@@ -22,6 +23,7 @@ type Pod struct {
 	Status              string            `json:"status"`
 	AppLabel            bool              `json:"appLabel"`
 	VersionLabel        bool              `json:"versionLabel"`
+	CustomDashboards    []string          `json:"-"` // not used in the UI but needed to build app and workload models
```
Comment (lucasponce) [inline models/pod.go]: Minor: If it's harmless I'm +1 to expose it into the json, perhaps it's useful for debugging purposes to fetch the workload's pods and check if things are ok.

**Comment by lucasponce** [comment]
Local testing, after following steps on https://issues.jboss.org/browse/KIALI-2054 I have:

```
GET https://admin:admin@kiali-istio-system.127.0.0.1.nip.io/api/namespaces/mesh-arena/apps/ui/customdashboard/vertx

{
  "title": "Vert.x",
  "charts": [
    {
      "name": "Server response time",
      "unit": "s",
      "spans": 6,
      "counterRate": null,
      "histogram": {
        "avg": {
          "matrix": []
        }
      }
    },
    {
      "name": "Client response time",
      "unit": "s",
      "spans": 6,
      "counterRate": null,
      "histogram": {
        "avg": {
          "matrix": []
        }
      }
    },
    {
      "name": "Server request count rate",
      "unit": "ops",
      "spans": 6,
      "counterRate": {
        "matrix": []
      },
      "histogram": null
    }
  ],
  "aggregations": [
    {
      "label": "code",
      "displayName": "Error code"
    },
    {
      "label": "path",
      "displayName": "Path"
    }
  ]
}

```

I don't have data as I didn't play with the mesh-arena demo, but I guess the result is correct.

File: models/pod.go (hunk 3 above)
Comment (jotak) [reply models/pod.go]: ok i'll do it

**Pull request merged**